edit history: Open the modal after fetching data.

matheuscteo opened the pull request:
The opening of the message edit history modal was moved to `fetch_and_render_message_history`, preventing the dialog to [flicker](https://github.com/zulip/zulip/issues/26582) with an empty entry. For completion the fetching function was moved to `initialize`.

<!-- Describe your pull request here.-->

In this PR the function `show_history` was moved to be called inside the `fetch_and_render_message_history`, in order to guarantee that the dialog only opens after the message history was completely fetched. `show_history` was also changed to open the edit message history dialog only if such dialog is not yet opened, keeping the dynamic elements.

Fixes: Solves issue [26582](https://github.com/zulip/zulip/issues/26582)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

sahil839 commented:
Thanks for working on this @matheuscteo. One bug that I noticed is that an empty modal is opened when using the "H" hotkey, so you would need to fix that.

Also, we might need to think about how to handle the case when server returns error while fetching the message history. Previously we used to show the error inside the modal itself, but now after the changes in this PR, the modal is only opened after data is fetched successfully.

Also please fix the commit message to follow the [commit message guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages) specifically the line-wrapping and adding "Fixes .." line.

sahil839 commented on web/src/message_edit_history.js:
```diff
@@ -190,7 +190,7 @@ export function initialize() {
         if (page_params.realm_allow_edit_history) {
-            show_history(message);
+            fetch_and_render_message_history(message);
             $("#message-history-cancel").trigger("focus");
```
I checked that this `trigger` code is not needed anymore as there is no such element and focus is handled by passing `focus_on_submit` as true to `dialog_widget`, so we can remove this in a separate commit.

timabbott commented:
@matheuscteo will you be able to finish this PR?

matheuscteo commented:
> @matheuscteo will you be able to finish this PR?

I'll try to address the comments this weekend but can't really guarantee. Sorry about that.
Thanks for adding the completion candidate tag for it.

matheuscteo commented:
Thanks for the feedback, @sahil839 . I've fixed the bug and removed the triggers, as you pointed out. I also have fixed the bug regarding the displaying of server errors. I'd be open to more feedback.

cc: @timabbott

sahil839 reviewed (commented):
Thanks for updating the PR. Posted a few comments. You should fix the commit history - there should only be two commits, one to open the modal after fetching data and one to remove the unnecessary tirggers.

Also, please the fix commit messages. There are some typos and formatting issues in first commit and you should clearly mention why we are removing the triggers in the "Remove unnecessary triggers" commit instead of just mentioning that it was pointed out in the PR. 

I would suggest you to read https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html to get a more clear idea of how Zulip expects commit history and commit messages to be.

ecxtacy commented:
@matheuscteo, are you still working on this ? I'd like to assist you with its completion. :smiley:

timabbott commented:
Closing, since https://github.com/zulip/zulip/issues/26582 was closed as no longer planned.
